relay-server/src/actors/project.rs

```diff
@@ -399,12 +403,22 @@ pub struct Project {
 
 impl Project {
     /// Creates a new `Project`.
-    pub fn new(key: ProjectKey, config: Arc<Config>) -> Self {
+    pub fn new(
+        key: ProjectKey,
+        config: Arc<Config>,
+        project_cache: Addr<ProjectCache>,
+        envelope_manager: Addr<EnvelopeManager>,
+        #[cfg(feature = "processing")] envelope_processor: Addr<EnvelopeProcessor>,
+    ) -> Self {
         Project {
             backoff: RetryBackoff::new(config.http_max_retry_interval()),
             next_fetch_attempt: None,
             last_updated_at: Instant::now(),
             project_key: key,
+            project_cache,
+            #[cfg(feature = "processing")]
+            envelope_processor,
+            envelope_manager,
```

Comment:
Out of scope for this PR, but the necessity of passing 3 `Addr`s into every single `Project` leads me to believe that we have a flaw in our design. Maybe we need to move even more functionality from `Project` to `ProjectCache`?

When we are done with https://github.com/getsentry/relay/issues/1818, we should probably draw a diagram of the current data flow, and see how we can improve it.